graylog2-web-interface/src/routing/Routes.ts

```diff
@@ -337,6 +338,26 @@ const pluginRoute = (routeKey, throwError = true) => {
 
 const getPluginRoute = (routeKey) => pluginRoute(routeKey, false);
 
+/**
+ * Checks if a plugin and its corresponding route is registered to the PluginStore
+ *
+ * @param {string} route
+ */
+function checkForRoute(route: string): boolean {
+  const pluginExports = PluginStore.exports('navigation');
+
+  if (!pluginExports) return false;
+
+  return !pluginExports.find((value) => value.description?.toLowerCase() === route);
+}
+
+/**
+ * Exported constants for use in seeing if the plugin and menu item are registered for addon components.
+ *
+ */
+export const enterpriseMenuIsMissing = checkForRoute('analyst tools');
+export const securityMenuIsMissing = checkForRoute('enterprise');
+
```

Comment:
Few suggestions regarding the naming:

```
function pluginMenuItemExists(description: string): boolean {
  const pluginMenuItems = PluginStore.exports('navigation');

  if (!pluginMenuItems) return false;

  return !pluginMenuItems.find(value) => value?.toLowerCase() === description.toLowerCase());
}
```

- `checkForRoute(route)` is a little bit misleading. Someone who is not familiar with the context will think this is a helper function for routes like '/the/route'. They will not understand that it is about plugable menu items.
- I implemented `description.toLowerCase()` to ensure we do not have a problem when some is not remembering that we expect a lowercase input.
- It looks like we need to implement `!checkForRoute()` (or now !menuItemExists()) for `enterpriseMenuIsMissing` and `securityMenuIsMissing`. This bug will hopefully be easier to catch with the new name, when reading the code.
- I know I suggested to place this code in the `Routes` file, but after having another look my preference is to only define:
```
const ENTERPRISE_ROUTE_DESCRIPTION = 'Enterprise';
const SECURITY_ROUTE_DESCRIPTION = 'Analyst Tools';
```
in this file and move the helper function `pluginMenuItemExists` to the `Navigation` file, since it is more related to the navigation than the actual routes.

Reply:
I couldn't reproduce 
> It looks like we need to implement !checkForRoute() (or now !menuItemExists()) for enterpriseMenuIsMissing and securityMenuIsMissing. This bug will hopefully be easier to catch with the new name, when reading the code.

The rest of these issues are addressed.

Reply:
Ok I think I got this to the correct state now.